core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -55,49 +78,77 @@
     /**
      * Constructs a new term query.
      *
-     * @param name  The name of the field
+     * @param fieldName  The name of the field
      * @param value The value of the term
      */
-    public FuzzyQueryBuilder(String name, Object value) {
-        this.name = name;
-        this.value = value;
+    public FuzzyQueryBuilder(String fieldName, Object value) {
```

Comment:
If we already break the builder API by changing the value-getter, it would be great to somehow also restrict the constructor to not accept every value type anymore. Maybe deprecate this constructor and instroduce new ones for String and Number? Not sure if also Date type is handled correctly when parsed to String here. 
In general I'm unsure here to how far this PR should change the status quo. I think only selected Object types worked here in the past, it should be stated somehere (docs?) what actually works. In our own code we really only use the constructor with String argument, so at least internally we could easily change that, but not sure how many client code this would break.